// @osmonauts/telescope/main/commands/transpile.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _prompt = require("../prompt");

var _index = _interopRequireDefault(require("../index"));

var _default = /*#__PURE__*/function () {
  var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(argv) {
    var questions, _yield$prompt, protoDirs, outPath, includeAminos, includeLCDClient;

    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            questions = [{
              _: true,
              type: 'path',
              name: 'protoDirs',
              message: 'where is the proto directory?',
              "default": './proto'
            }, {
              _: true,
              type: 'path',
              name: 'outPath',
              message: 'where is the output directory?',
              "default": './src/proto'
            }, {
              type: 'confirm',
              name: 'includeAminos',
              message: 'output amino messages?',
              "default": true
            }, {
              type: 'confirm',
              name: 'includeLCDClient',
              message: 'output LCD Clients?',
              "default": false
            }];
            _context.next = 3;
            return (0, _prompt.prompt)(questions, argv);

          case 3:
            _yield$prompt = _context.sent;
            protoDirs = _yield$prompt.protoDirs;
            outPath = _yield$prompt.outPath;
            includeAminos = _yield$prompt.includeAminos;
            includeLCDClient = _yield$prompt.includeLCDClient;

            if (!Array.isArray(protoDirs)) {
              protoDirs = [protoDirs];
            }

            (0, _index["default"])({
              protoDirs: protoDirs,
              outPath: outPath,
              options: {
                includeAminos: includeAminos,
                includeLCDClient: includeLCDClient
              }
            });
            console.log("\u2728 transpilation successful!");

          case 11:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));

  return function (_x) {
    return _ref.apply(this, arguments);
  };
}();

exports["default"] = _default;
