// akashjs/main/proto/akash/cert/v1beta2/cert.amino.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AminoConverter = void 0;
var AminoConverter = {
  "/akash.cert.v1beta2.MsgCreateCertificate": {
    aminoType: "/akash.cert.v1beta2.MsgCreateCertificate",
    toAmino: function toAmino(_ref) {
      var owner = _ref.owner,
          cert = _ref.cert,
          pubkey = _ref.pubkey;
      return {
        owner: owner,
        cert: cert,
        pubkey: pubkey
      };
    },
    fromAmino: function fromAmino(_ref2) {
      var owner = _ref2.owner,
          cert = _ref2.cert,
          pubkey = _ref2.pubkey;
      return {
        owner: owner,
        cert: cert,
        pubkey: pubkey
      };
    }
  },
  "/akash.cert.v1beta2.MsgRevokeCertificate": {
    aminoType: "/akash.cert.v1beta2.MsgRevokeCertificate",
    toAmino: function toAmino(_ref3) {
      var id = _ref3.id;
      return {
        id: {
          owner: id.owner,
          serial: id.serial
        }
      };
    },
    fromAmino: function fromAmino(_ref4) {
      var id = _ref4.id;
      return {
        id: {
          owner: id.owner,
          serial: id.serial
        }
      };
    }
  }
};
exports.AminoConverter = AminoConverter;
