// akashjs/main/proto/akash/cert/v1beta2/cert.registry.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.registry = exports.load = exports.MessageComposer = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _cert = require("./cert");

var registry = [["/akash.cert.v1beta2.MsgCreateCertificate", _cert.MsgCreateCertificate], ["/akash.cert.v1beta2.MsgRevokeCertificate", _cert.MsgRevokeCertificate]];
exports.registry = registry;

var load = function load(protoRegistry) {
  registry.forEach(function (_ref) {
    var _ref2 = (0, _slicedToArray2["default"])(_ref, 2),
        typeUrl = _ref2[0],
        mod = _ref2[1];

    protoRegistry.register(typeUrl, mod);
  });
};

exports.load = load;
var MessageComposer = {
  encoded: {
    createCertificate: function createCertificate(value) {
      return {
        type_url: "/akash.cert.v1beta2.MsgCreateCertificate",
        value: _cert.MsgCreateCertificate.encode(value).finish()
      };
    },
    revokeCertificate: function revokeCertificate(value) {
      return {
        type_url: "/akash.cert.v1beta2.MsgRevokeCertificate",
        value: _cert.MsgRevokeCertificate.encode(value).finish()
      };
    }
  },
  withTypeUrl: {
    createCertificate: function createCertificate(value) {
      return {
        typeUrl: "/akash.cert.v1beta2.MsgCreateCertificate",
        value: value
      };
    },
    revokeCertificate: function revokeCertificate(value) {
      return {
        typeUrl: "/akash.cert.v1beta2.MsgRevokeCertificate",
        value: value
      };
    }
  },
  toJSON: {
    createCertificate: function createCertificate(value) {
      return {
        typeUrl: "/akash.cert.v1beta2.MsgCreateCertificate",
        value: _cert.MsgCreateCertificate.toJSON(value)
      };
    },
    revokeCertificate: function revokeCertificate(value) {
      return {
        typeUrl: "/akash.cert.v1beta2.MsgRevokeCertificate",
        value: _cert.MsgRevokeCertificate.toJSON(value)
      };
    }
  },
  fromJSON: {
    createCertificate: function createCertificate(value) {
      return {
        typeUrl: "/akash.cert.v1beta2.MsgCreateCertificate",
        value: _cert.MsgCreateCertificate.fromJSON(value)
      };
    },
    revokeCertificate: function revokeCertificate(value) {
      return {
        typeUrl: "/akash.cert.v1beta2.MsgRevokeCertificate",
        value: _cert.MsgRevokeCertificate.fromJSON(value)
      };
    }
  },
  fromPartial: {
    createCertificate: function createCertificate(value) {
      return {
        typeUrl: "/akash.cert.v1beta2.MsgCreateCertificate",
        value: _cert.MsgCreateCertificate.fromPartial(value)
      };
    },
    revokeCertificate: function revokeCertificate(value) {
      return {
        typeUrl: "/akash.cert.v1beta2.MsgRevokeCertificate",
        value: _cert.MsgRevokeCertificate.fromPartial(value)
      };
    }
  }
};
exports.MessageComposer = MessageComposer;
