// wasm-ast-types/main/utils/types.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getType = exports.getPropertyType = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var t = _interopRequireWildcard(require("@babel/types"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var getTypeFromRef = function getTypeFromRef($ref) {
  switch ($ref) {
    case '#/definitions/Binary':
      return t.tsTypeReference(t.identifier('Binary'));

    default:
      if ($ref.startsWith('#/definitions/')) {
        return t.tsTypeReference(t.identifier($ref.replace('#/definitions/', '')));
      }

      throw new Error('what is $ref: ' + $ref);
  }
};

var getArrayTypeFromRef = function getArrayTypeFromRef($ref) {
  return t.tsArrayType(getTypeFromRef($ref));
};

var getArrayTypeFromType = function getArrayTypeFromType(type) {
  return t.tsArrayType(getType(type));
};

var getType = function getType(type) {
  switch (type) {
    case 'string':
      return t.tsStringKeyword();

    case 'boolean':
      return t.tSBooleanKeyword();

    case 'integer':
      return t.tsNumberKeyword();

    default:
      throw new Error('contact maintainers [unknown type]: ' + type);
  }
};

exports.getType = getType;

var getPropertyType = function getPropertyType(schema, prop) {
  var _schema$properties, _schema$required, _schema$required2;

  var props = (_schema$properties = schema.properties) !== null && _schema$properties !== void 0 ? _schema$properties : {};
  var info = props[prop];
  var type = null;
  var optional = (_schema$required = schema.required) === null || _schema$required === void 0 ? void 0 : _schema$required.includes(prop);

  if (info.allOf && info.allOf.length === 1) {
    info = info.allOf[0];
  }

  if (typeof info.$ref === 'string') {
    type = getTypeFromRef(info.$ref);
  }

  if (Array.isArray(info.anyOf)) {
    // assuming 2nd is null, but let's check to ensure
    if (info.anyOf.length !== 2) {
      throw new Error('case not handled by transpiler. contact maintainers.');
    }

    var _info$anyOf = (0, _slicedToArray2["default"])(info.anyOf, 2),
        nullableType = _info$anyOf[0],
        nullType = _info$anyOf[1];

    if ((nullType === null || nullType === void 0 ? void 0 : nullType.type) !== 'null') {
      throw new Error('case not handled by transpiler. contact maintainers.');
    }

    type = getTypeFromRef(nullableType === null || nullableType === void 0 ? void 0 : nullableType.$ref);
    optional = true;
  }

  if (typeof info.type === 'string') {
    if (info.type === 'array') {
      if (info.items.$ref) {
        type = getArrayTypeFromRef(info.items.$ref);
      } else {
        type = getArrayTypeFromType(info.items.type);
      }
    } else {
      type = getType(info.type);
    }
  }

  if (Array.isArray(info.type)) {
    // assuming 2nd is null, but let's check to ensure
    if (info.type.length !== 2) {
      throw new Error('please report this to maintainers (field type): ' + JSON.stringify(info, null, 2));
    }

    var _info$type = (0, _slicedToArray2["default"])(info.type, 2),
        _nullableType = _info$type[0],
        _nullType = _info$type[1];

    if (_nullType !== 'null') {
      throw new Error('please report this to maintainers (field type): ' + JSON.stringify(info, null, 2));
    }

    if (_nullableType === 'array') {
      type = t.tsArrayType(getType(info.items.type));
    } else {
      type = getType(_nullableType);
    }

    optional = true;
  }

  if (!type) {
    throw new Error('cannot find type for ' + JSON.stringify(info));
  }

  if ((_schema$required2 = schema.required) !== null && _schema$required2 !== void 0 && _schema$required2.includes(prop)) {
    optional = false;
  }

  return {
    type: type,
    optional: optional
  };
};

exports.getPropertyType = getPropertyType;
