// akashjs/main/proto/akash/market/v1beta2/bid.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MsgCreateBidResponse = exports.MsgCreateBid = exports.MsgCloseBidResponse = exports.MsgCloseBid = exports.Bid_State = exports.BidID = exports.BidFilters = exports.Bid = void 0;
exports.bid_StateFromJSON = bid_StateFromJSON;
exports.bid_StateToJSON = bid_StateToJSON;

var _order = require("./order");

var _coin = require("../../../cosmos/base/v1beta1/coin");

var _m0 = _interopRequireWildcard(require("protobufjs/minimal"));

var _helpers = require("@osmonauts/helpers");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/** State is an enum which refers to state of bid */
var Bid_State;
exports.Bid_State = Bid_State;

(function (Bid_State) {
  Bid_State[Bid_State["invalid"] = 0] = "invalid";
  Bid_State[Bid_State["open"] = 1] = "open";
  Bid_State[Bid_State["active"] = 2] = "active";
  Bid_State[Bid_State["lost"] = 3] = "lost";
  Bid_State[Bid_State["closed"] = 4] = "closed";
  Bid_State[Bid_State["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Bid_State || (exports.Bid_State = Bid_State = {}));

function bid_StateFromJSON(object) {
  switch (object) {
    case 0:
    case "invalid":
      return Bid_State.invalid;

    case 1:
    case "open":
      return Bid_State.open;

    case 2:
    case "active":
      return Bid_State.active;

    case 3:
    case "lost":
      return Bid_State.lost;

    case 4:
    case "closed":
      return Bid_State.closed;

    case -1:
    case "UNRECOGNIZED":
    default:
      return Bid_State.UNRECOGNIZED;
  }
}

function bid_StateToJSON(object) {
  switch (object) {
    case Bid_State.invalid:
      return "invalid";

    case Bid_State.open:
      return "open";

    case Bid_State.active:
      return "active";

    case Bid_State.lost:
      return "lost";

    case Bid_State.closed:
      return "closed";

    default:
      return "UNKNOWN";
  }
}
/** MsgCreateBid defines an SDK message for creating Bid */


function createBaseMsgCreateBid() {
  return {
    order: undefined,
    provider: "",
    price: undefined,
    deposit: undefined
  };
}

var MsgCreateBid = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.order !== undefined) {
      _order.OrderID.encode(message.order, writer.uint32(10).fork()).ldelim();
    }

    if (message.provider !== "") {
      writer.uint32(18).string(message.provider);
    }

    if (message.price !== undefined) {
      _coin.DecCoin.encode(message.price, writer.uint32(26).fork()).ldelim();
    }

    if (message.deposit !== undefined) {
      _coin.Coin.encode(message.deposit, writer.uint32(34).fork()).ldelim();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseMsgCreateBid();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.order = _order.OrderID.decode(reader, reader.uint32());
          break;

        case 2:
          message.provider = reader.string();
          break;

        case 3:
          message.price = _coin.DecCoin.decode(reader, reader.uint32());
          break;

        case 4:
          message.deposit = _coin.Coin.decode(reader, reader.uint32());
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      order: (0, _helpers.isSet)(object.order) ? _order.OrderID.fromJSON(object.order) : undefined,
      provider: (0, _helpers.isSet)(object.provider) ? String(object.provider) : "",
      price: (0, _helpers.isSet)(object.price) ? _coin.DecCoin.fromJSON(object.price) : undefined,
      deposit: (0, _helpers.isSet)(object.deposit) ? _coin.Coin.fromJSON(object.deposit) : undefined
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.order !== undefined && (obj.order = message.order ? _order.OrderID.toJSON(message.order) : undefined);
    message.provider !== undefined && (obj.provider = message.provider);
    message.price !== undefined && (obj.price = message.price ? _coin.DecCoin.toJSON(message.price) : undefined);
    message.deposit !== undefined && (obj.deposit = message.deposit ? _coin.Coin.toJSON(message.deposit) : undefined);
    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$provider;

    var message = createBaseMsgCreateBid();
    message.order = object.order !== undefined && object.order !== null ? _order.OrderID.fromPartial(object.order) : undefined;
    message.provider = (_object$provider = object.provider) !== null && _object$provider !== void 0 ? _object$provider : "";
    message.price = object.price !== undefined && object.price !== null ? _coin.DecCoin.fromPartial(object.price) : undefined;
    message.deposit = object.deposit !== undefined && object.deposit !== null ? _coin.Coin.fromPartial(object.deposit) : undefined;
    return message;
  }
};
exports.MsgCreateBid = MsgCreateBid;

function createBaseMsgCreateBidResponse() {
  return {};
}

var MsgCreateBidResponse = {
  encode: function encode(_) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();
    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseMsgCreateBidResponse();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(_) {
    return {};
  },
  toJSON: function toJSON(_) {
    var obj = {};
    return obj;
  },
  fromPartial: function fromPartial(_) {
    var message = createBaseMsgCreateBidResponse();
    return message;
  }
};
exports.MsgCreateBidResponse = MsgCreateBidResponse;

function createBaseMsgCloseBid() {
  return {
    bidId: undefined
  };
}

var MsgCloseBid = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.bidId !== undefined) {
      BidID.encode(message.bidId, writer.uint32(10).fork()).ldelim();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseMsgCloseBid();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.bidId = BidID.decode(reader, reader.uint32());
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      bidId: (0, _helpers.isSet)(object.bidId) ? BidID.fromJSON(object.bidId) : undefined
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.bidId !== undefined && (obj.bidId = message.bidId ? BidID.toJSON(message.bidId) : undefined);
    return obj;
  },
  fromPartial: function fromPartial(object) {
    var message = createBaseMsgCloseBid();
    message.bidId = object.bidId !== undefined && object.bidId !== null ? BidID.fromPartial(object.bidId) : undefined;
    return message;
  }
};
exports.MsgCloseBid = MsgCloseBid;

function createBaseMsgCloseBidResponse() {
  return {};
}

var MsgCloseBidResponse = {
  encode: function encode(_) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();
    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseMsgCloseBidResponse();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(_) {
    return {};
  },
  toJSON: function toJSON(_) {
    var obj = {};
    return obj;
  },
  fromPartial: function fromPartial(_) {
    var message = createBaseMsgCloseBidResponse();
    return message;
  }
};
exports.MsgCloseBidResponse = MsgCloseBidResponse;

function createBaseBidID() {
  return {
    owner: "",
    dseq: _helpers.Long.UZERO,
    gseq: 0,
    oseq: 0,
    provider: ""
  };
}

var BidID = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.owner !== "") {
      writer.uint32(10).string(message.owner);
    }

    if (!message.dseq.isZero()) {
      writer.uint32(16).uint64(message.dseq);
    }

    if (message.gseq !== 0) {
      writer.uint32(24).uint32(message.gseq);
    }

    if (message.oseq !== 0) {
      writer.uint32(32).uint32(message.oseq);
    }

    if (message.provider !== "") {
      writer.uint32(42).string(message.provider);
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseBidID();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.owner = reader.string();
          break;

        case 2:
          message.dseq = reader.uint64();
          break;

        case 3:
          message.gseq = reader.uint32();
          break;

        case 4:
          message.oseq = reader.uint32();
          break;

        case 5:
          message.provider = reader.string();
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      owner: (0, _helpers.isSet)(object.owner) ? String(object.owner) : "",
      dseq: (0, _helpers.isSet)(object.dseq) ? _helpers.Long.fromString(object.dseq) : _helpers.Long.UZERO,
      gseq: (0, _helpers.isSet)(object.gseq) ? Number(object.gseq) : 0,
      oseq: (0, _helpers.isSet)(object.oseq) ? Number(object.oseq) : 0,
      provider: (0, _helpers.isSet)(object.provider) ? String(object.provider) : ""
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.owner !== undefined && (obj.owner = message.owner);
    message.dseq !== undefined && (obj.dseq = (message.dseq || _helpers.Long.UZERO).toString());
    message.gseq !== undefined && (obj.gseq = Math.round(message.gseq));
    message.oseq !== undefined && (obj.oseq = Math.round(message.oseq));
    message.provider !== undefined && (obj.provider = message.provider);
    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$owner, _object$gseq, _object$oseq, _object$provider2;

    var message = createBaseBidID();
    message.owner = (_object$owner = object.owner) !== null && _object$owner !== void 0 ? _object$owner : "";
    message.dseq = object.dseq !== undefined && object.dseq !== null ? _helpers.Long.fromValue(object.dseq) : _helpers.Long.UZERO;
    message.gseq = (_object$gseq = object.gseq) !== null && _object$gseq !== void 0 ? _object$gseq : 0;
    message.oseq = (_object$oseq = object.oseq) !== null && _object$oseq !== void 0 ? _object$oseq : 0;
    message.provider = (_object$provider2 = object.provider) !== null && _object$provider2 !== void 0 ? _object$provider2 : "";
    return message;
  }
};
exports.BidID = BidID;

function createBaseBid() {
  return {
    bidId: undefined,
    state: 0,
    price: undefined,
    createdAt: _helpers.Long.ZERO
  };
}

var Bid = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.bidId !== undefined) {
      BidID.encode(message.bidId, writer.uint32(10).fork()).ldelim();
    }

    if (message.state !== 0) {
      writer.uint32(16).int32(message.state);
    }

    if (message.price !== undefined) {
      _coin.DecCoin.encode(message.price, writer.uint32(26).fork()).ldelim();
    }

    if (!message.createdAt.isZero()) {
      writer.uint32(32).int64(message.createdAt);
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseBid();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.bidId = BidID.decode(reader, reader.uint32());
          break;

        case 2:
          message.state = reader.int32();
          break;

        case 3:
          message.price = _coin.DecCoin.decode(reader, reader.uint32());
          break;

        case 4:
          message.createdAt = reader.int64();
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      bidId: (0, _helpers.isSet)(object.bidId) ? BidID.fromJSON(object.bidId) : undefined,
      state: (0, _helpers.isSet)(object.state) ? bid_StateFromJSON(object.state) : 0,
      price: (0, _helpers.isSet)(object.price) ? _coin.DecCoin.fromJSON(object.price) : undefined,
      createdAt: (0, _helpers.isSet)(object.createdAt) ? _helpers.Long.fromString(object.createdAt) : _helpers.Long.ZERO
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.bidId !== undefined && (obj.bidId = message.bidId ? BidID.toJSON(message.bidId) : undefined);
    message.state !== undefined && (obj.state = bid_StateToJSON(message.state));
    message.price !== undefined && (obj.price = message.price ? _coin.DecCoin.toJSON(message.price) : undefined);
    message.createdAt !== undefined && (obj.createdAt = (message.createdAt || _helpers.Long.ZERO).toString());
    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$state;

    var message = createBaseBid();
    message.bidId = object.bidId !== undefined && object.bidId !== null ? BidID.fromPartial(object.bidId) : undefined;
    message.state = (_object$state = object.state) !== null && _object$state !== void 0 ? _object$state : 0;
    message.price = object.price !== undefined && object.price !== null ? _coin.DecCoin.fromPartial(object.price) : undefined;
    message.createdAt = object.createdAt !== undefined && object.createdAt !== null ? _helpers.Long.fromValue(object.createdAt) : _helpers.Long.ZERO;
    return message;
  }
};
exports.Bid = Bid;

function createBaseBidFilters() {
  return {
    owner: "",
    dseq: _helpers.Long.UZERO,
    gseq: 0,
    oseq: 0,
    provider: "",
    state: ""
  };
}

var BidFilters = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.owner !== "") {
      writer.uint32(10).string(message.owner);
    }

    if (!message.dseq.isZero()) {
      writer.uint32(16).uint64(message.dseq);
    }

    if (message.gseq !== 0) {
      writer.uint32(24).uint32(message.gseq);
    }

    if (message.oseq !== 0) {
      writer.uint32(32).uint32(message.oseq);
    }

    if (message.provider !== "") {
      writer.uint32(42).string(message.provider);
    }

    if (message.state !== "") {
      writer.uint32(50).string(message.state);
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseBidFilters();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.owner = reader.string();
          break;

        case 2:
          message.dseq = reader.uint64();
          break;

        case 3:
          message.gseq = reader.uint32();
          break;

        case 4:
          message.oseq = reader.uint32();
          break;

        case 5:
          message.provider = reader.string();
          break;

        case 6:
          message.state = reader.string();
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      owner: (0, _helpers.isSet)(object.owner) ? String(object.owner) : "",
      dseq: (0, _helpers.isSet)(object.dseq) ? _helpers.Long.fromString(object.dseq) : _helpers.Long.UZERO,
      gseq: (0, _helpers.isSet)(object.gseq) ? Number(object.gseq) : 0,
      oseq: (0, _helpers.isSet)(object.oseq) ? Number(object.oseq) : 0,
      provider: (0, _helpers.isSet)(object.provider) ? String(object.provider) : "",
      state: (0, _helpers.isSet)(object.state) ? String(object.state) : ""
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.owner !== undefined && (obj.owner = message.owner);
    message.dseq !== undefined && (obj.dseq = (message.dseq || _helpers.Long.UZERO).toString());
    message.gseq !== undefined && (obj.gseq = Math.round(message.gseq));
    message.oseq !== undefined && (obj.oseq = Math.round(message.oseq));
    message.provider !== undefined && (obj.provider = message.provider);
    message.state !== undefined && (obj.state = message.state);
    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$owner2, _object$gseq2, _object$oseq2, _object$provider3, _object$state2;

    var message = createBaseBidFilters();
    message.owner = (_object$owner2 = object.owner) !== null && _object$owner2 !== void 0 ? _object$owner2 : "";
    message.dseq = object.dseq !== undefined && object.dseq !== null ? _helpers.Long.fromValue(object.dseq) : _helpers.Long.UZERO;
    message.gseq = (_object$gseq2 = object.gseq) !== null && _object$gseq2 !== void 0 ? _object$gseq2 : 0;
    message.oseq = (_object$oseq2 = object.oseq) !== null && _object$oseq2 !== void 0 ? _object$oseq2 : 0;
    message.provider = (_object$provider3 = object.provider) !== null && _object$provider3 !== void 0 ? _object$provider3 : "";
    message.state = (_object$state2 = object.state) !== null && _object$state2 !== void 0 ? _object$state2 : "";
    return message;
  }
};
exports.BidFilters = BidFilters;
