// akashjs/main/proto/akash/market/v1beta2/service.amino.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AminoConverter = void 0;

var _helpers = require("@osmonauts/helpers");

var AminoConverter = {
  "/akash.market.v1beta2.MsgCreateBid": {
    aminoType: "/akash.market.v1beta2.MsgCreateBid",
    toAmino: function toAmino(_ref) {
      var order = _ref.order,
          provider = _ref.provider,
          price = _ref.price,
          deposit = _ref.deposit;
      return {
        order: {
          owner: order.owner,
          dseq: order.dseq.toString(),
          gseq: order.gseq,
          oseq: order.oseq
        },
        provider: provider,
        price: {
          denom: price.denom,
          amount: price.amount
        },
        deposit: {
          denom: deposit.denom,
          amount: _helpers.Long.fromNumber(deposit.amount).toString()
        }
      };
    },
    fromAmino: function fromAmino(_ref2) {
      var order = _ref2.order,
          provider = _ref2.provider,
          price = _ref2.price,
          deposit = _ref2.deposit;
      return {
        order: {
          owner: order.owner,
          dseq: _helpers.Long.fromString(order.dseq),
          gseq: order.gseq,
          oseq: order.oseq
        },
        provider: provider,
        price: {
          denom: price.denom,
          amount: price.amount
        },
        deposit: {
          denom: deposit.denom,
          amount: deposit.amount
        }
      };
    }
  },
  "/akash.market.v1beta2.MsgCloseBid": {
    aminoType: "/akash.market.v1beta2.MsgCloseBid",
    toAmino: function toAmino(_ref3) {
      var bidId = _ref3.bidId;
      return {
        bid_id: {
          owner: bidId.owner,
          dseq: bidId.dseq.toString(),
          gseq: bidId.gseq,
          oseq: bidId.oseq,
          provider: bidId.provider
        }
      };
    },
    fromAmino: function fromAmino(_ref4) {
      var bid_id = _ref4.bid_id;
      return {
        bidId: {
          owner: bid_id.owner,
          dseq: _helpers.Long.fromString(bid_id.dseq),
          gseq: bid_id.gseq,
          oseq: bid_id.oseq,
          provider: bid_id.provider
        }
      };
    }
  },
  "/akash.market.v1beta2.MsgWithdrawLease": {
    aminoType: "/akash.market.v1beta2.MsgWithdrawLease",
    toAmino: function toAmino(_ref5) {
      var bidId = _ref5.bidId;
      return {
        bid_id: {
          owner: bidId.owner,
          dseq: bidId.dseq.toString(),
          gseq: bidId.gseq,
          oseq: bidId.oseq,
          provider: bidId.provider
        }
      };
    },
    fromAmino: function fromAmino(_ref6) {
      var bid_id = _ref6.bid_id;
      return {
        bidId: {
          owner: bid_id.owner,
          dseq: _helpers.Long.fromString(bid_id.dseq),
          gseq: bid_id.gseq,
          oseq: bid_id.oseq,
          provider: bid_id.provider
        }
      };
    }
  },
  "/akash.market.v1beta2.MsgCreateLease": {
    aminoType: "/akash.market.v1beta2.MsgCreateLease",
    toAmino: function toAmino(_ref7) {
      var bidId = _ref7.bidId;
      return {
        bid_id: {
          owner: bidId.owner,
          dseq: bidId.dseq.toString(),
          gseq: bidId.gseq,
          oseq: bidId.oseq,
          provider: bidId.provider
        }
      };
    },
    fromAmino: function fromAmino(_ref8) {
      var bid_id = _ref8.bid_id;
      return {
        bidId: {
          owner: bid_id.owner,
          dseq: _helpers.Long.fromString(bid_id.dseq),
          gseq: bid_id.gseq,
          oseq: bid_id.oseq,
          provider: bid_id.provider
        }
      };
    }
  },
  "/akash.market.v1beta2.MsgCloseLease": {
    aminoType: "/akash.market.v1beta2.MsgCloseLease",
    toAmino: function toAmino(_ref9) {
      var leaseId = _ref9.leaseId;
      return {
        lease_id: {
          owner: leaseId.owner,
          dseq: leaseId.dseq.toString(),
          gseq: leaseId.gseq,
          oseq: leaseId.oseq,
          provider: leaseId.provider
        }
      };
    },
    fromAmino: function fromAmino(_ref10) {
      var lease_id = _ref10.lease_id;
      return {
        leaseId: {
          owner: lease_id.owner,
          dseq: _helpers.Long.fromString(lease_id.dseq),
          gseq: lease_id.gseq,
          oseq: lease_id.oseq,
          provider: lease_id.provider
        }
      };
    }
  }
};
exports.AminoConverter = AminoConverter;
