// akashjs/main/proto/akash/inflation/v1beta2/params.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Params = void 0;

var _m0 = _interopRequireWildcard(require("protobufjs/minimal"));

var _helpers = require("@osmonauts/helpers");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function createBaseParams() {
  return {
    inflationDecayFactor: "",
    initialInflation: "",
    variance: ""
  };
}

var Params = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.inflationDecayFactor !== "") {
      writer.uint32(10).string(message.inflationDecayFactor);
    }

    if (message.initialInflation !== "") {
      writer.uint32(18).string(message.initialInflation);
    }

    if (message.variance !== "") {
      writer.uint32(26).string(message.variance);
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseParams();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.inflationDecayFactor = reader.string();
          break;

        case 2:
          message.initialInflation = reader.string();
          break;

        case 3:
          message.variance = reader.string();
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      inflationDecayFactor: (0, _helpers.isSet)(object.inflationDecayFactor) ? String(object.inflationDecayFactor) : "",
      initialInflation: (0, _helpers.isSet)(object.initialInflation) ? String(object.initialInflation) : "",
      variance: (0, _helpers.isSet)(object.variance) ? String(object.variance) : ""
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.inflationDecayFactor !== undefined && (obj.inflationDecayFactor = message.inflationDecayFactor);
    message.initialInflation !== undefined && (obj.initialInflation = message.initialInflation);
    message.variance !== undefined && (obj.variance = message.variance);
    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$inflationDeca, _object$initialInflat, _object$variance;

    var message = createBaseParams();
    message.inflationDecayFactor = (_object$inflationDeca = object.inflationDecayFactor) !== null && _object$inflationDeca !== void 0 ? _object$inflationDeca : "";
    message.initialInflation = (_object$initialInflat = object.initialInflation) !== null && _object$initialInflat !== void 0 ? _object$initialInflat : "";
    message.variance = (_object$variance = object.variance) !== null && _object$variance !== void 0 ? _object$variance : "";
    return message;
  }
};
exports.Params = Params;
