// akashjs/main/proto/akash/deployment/v1beta2/resource.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Resource = void 0;

var _resourceunits = require("../../base/v1beta2/resourceunits");

var _coin = require("../../../cosmos/base/v1beta1/coin");

var _m0 = _interopRequireWildcard(require("protobufjs/minimal"));

var _helpers = require("@osmonauts/helpers");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function createBaseResource() {
  return {
    resources: undefined,
    count: 0,
    price: undefined
  };
}

var Resource = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.resources !== undefined) {
      _resourceunits.ResourceUnits.encode(message.resources, writer.uint32(10).fork()).ldelim();
    }

    if (message.count !== 0) {
      writer.uint32(16).uint32(message.count);
    }

    if (message.price !== undefined) {
      _coin.DecCoin.encode(message.price, writer.uint32(26).fork()).ldelim();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseResource();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.resources = _resourceunits.ResourceUnits.decode(reader, reader.uint32());
          break;

        case 2:
          message.count = reader.uint32();
          break;

        case 3:
          message.price = _coin.DecCoin.decode(reader, reader.uint32());
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      resources: (0, _helpers.isSet)(object.resources) ? _resourceunits.ResourceUnits.fromJSON(object.resources) : undefined,
      count: (0, _helpers.isSet)(object.count) ? Number(object.count) : 0,
      price: (0, _helpers.isSet)(object.price) ? _coin.DecCoin.fromJSON(object.price) : undefined
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.resources !== undefined && (obj.resources = message.resources ? _resourceunits.ResourceUnits.toJSON(message.resources) : undefined);
    message.count !== undefined && (obj.count = Math.round(message.count));
    message.price !== undefined && (obj.price = message.price ? _coin.DecCoin.toJSON(message.price) : undefined);
    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$count;

    var message = createBaseResource();
    message.resources = object.resources !== undefined && object.resources !== null ? _resourceunits.ResourceUnits.fromPartial(object.resources) : undefined;
    message.count = (_object$count = object.count) !== null && _object$count !== void 0 ? _object$count : 0;
    message.price = object.price !== undefined && object.price !== null ? _coin.DecCoin.fromPartial(object.price) : undefined;
    return message;
  }
};
exports.Resource = Resource;
