// akashjs/module/proto/akash/market/v1beta2/service.amino.js
import { Long } from "@osmonauts/helpers";
export const AminoConverter = {
  "/akash.market.v1beta2.MsgCreateBid": {
    aminoType: "/akash.market.v1beta2.MsgCreateBid",
    toAmino: ({
      order,
      provider,
      price,
      deposit
    }) => {
      return {
        order: {
          owner: order.owner,
          dseq: order.dseq.toString(),
          gseq: order.gseq,
          oseq: order.oseq
        },
        provider,
        price: {
          denom: price.denom,
          amount: price.amount
        },
        deposit: {
          denom: deposit.denom,
          amount: Long.fromNumber(deposit.amount).toString()
        }
      };
    },
    fromAmino: ({
      order,
      provider,
      price,
      deposit
    }) => {
      return {
        order: {
          owner: order.owner,
          dseq: Long.fromString(order.dseq),
          gseq: order.gseq,
          oseq: order.oseq
        },
        provider,
        price: {
          denom: price.denom,
          amount: price.amount
        },
        deposit: {
          denom: deposit.denom,
          amount: deposit.amount
        }
      };
    }
  },
  "/akash.market.v1beta2.MsgCloseBid": {
    aminoType: "/akash.market.v1beta2.MsgCloseBid",
    toAmino: ({
      bidId
    }) => {
      return {
        bid_id: {
          owner: bidId.owner,
          dseq: bidId.dseq.toString(),
          gseq: bidId.gseq,
          oseq: bidId.oseq,
          provider: bidId.provider
        }
      };
    },
    fromAmino: ({
      bid_id
    }) => {
      return {
        bidId: {
          owner: bid_id.owner,
          dseq: Long.fromString(bid_id.dseq),
          gseq: bid_id.gseq,
          oseq: bid_id.oseq,
          provider: bid_id.provider
        }
      };
    }
  },
  "/akash.market.v1beta2.MsgWithdrawLease": {
    aminoType: "/akash.market.v1beta2.MsgWithdrawLease",
    toAmino: ({
      bidId
    }) => {
      return {
        bid_id: {
          owner: bidId.owner,
          dseq: bidId.dseq.toString(),
          gseq: bidId.gseq,
          oseq: bidId.oseq,
          provider: bidId.provider
        }
      };
    },
    fromAmino: ({
      bid_id
    }) => {
      return {
        bidId: {
          owner: bid_id.owner,
          dseq: Long.fromString(bid_id.dseq),
          gseq: bid_id.gseq,
          oseq: bid_id.oseq,
          provider: bid_id.provider
        }
      };
    }
  },
  "/akash.market.v1beta2.MsgCreateLease": {
    aminoType: "/akash.market.v1beta2.MsgCreateLease",
    toAmino: ({
      bidId
    }) => {
      return {
        bid_id: {
          owner: bidId.owner,
          dseq: bidId.dseq.toString(),
          gseq: bidId.gseq,
          oseq: bidId.oseq,
          provider: bidId.provider
        }
      };
    },
    fromAmino: ({
      bid_id
    }) => {
      return {
        bidId: {
          owner: bid_id.owner,
          dseq: Long.fromString(bid_id.dseq),
          gseq: bid_id.gseq,
          oseq: bid_id.oseq,
          provider: bid_id.provider
        }
      };
    }
  },
  "/akash.market.v1beta2.MsgCloseLease": {
    aminoType: "/akash.market.v1beta2.MsgCloseLease",
    toAmino: ({
      leaseId
    }) => {
      return {
        lease_id: {
          owner: leaseId.owner,
          dseq: leaseId.dseq.toString(),
          gseq: leaseId.gseq,
          oseq: leaseId.oseq,
          provider: leaseId.provider
        }
      };
    },
    fromAmino: ({
      lease_id
    }) => {
      return {
        leaseId: {
          owner: lease_id.owner,
          dseq: Long.fromString(lease_id.dseq),
          gseq: lease_id.gseq,
          oseq: lease_id.oseq,
          provider: lease_id.provider
        }
      };
    }
  }
};
