// akashjs/main/proto/akash/audit/v1beta1/audit.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Provider = exports.MsgSignProviderAttributesResponse = exports.MsgSignProviderAttributes = exports.MsgDeleteProviderAttributesResponse = exports.MsgDeleteProviderAttributes = exports.AuditedAttributes = exports.AttributesResponse = exports.AttributesFilters = void 0;

var _attribute = require("../../base/v1beta1/attribute");

var _m0 = _interopRequireWildcard(require("protobufjs/minimal"));

var _helpers = require("@osmonauts/helpers");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function createBaseProvider() {
  return {
    owner: "",
    auditor: "",
    attributes: []
  };
}

var Provider = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.owner !== "") {
      writer.uint32(10).string(message.owner);
    }

    if (message.auditor !== "") {
      writer.uint32(18).string(message.auditor);
    }

    var _iterator = _createForOfIteratorHelper(message.attributes),
        _step;

    try {
      for (_iterator.s(); !(_step = _iterator.n()).done;) {
        var v = _step.value;

        _attribute.Attribute.encode(v, writer.uint32(34).fork()).ldelim();
      }
    } catch (err) {
      _iterator.e(err);
    } finally {
      _iterator.f();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseProvider();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.owner = reader.string();
          break;

        case 2:
          message.auditor = reader.string();
          break;

        case 4:
          message.attributes.push(_attribute.Attribute.decode(reader, reader.uint32()));
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      owner: (0, _helpers.isSet)(object.owner) ? String(object.owner) : "",
      auditor: (0, _helpers.isSet)(object.auditor) ? String(object.auditor) : "",
      attributes: Array.isArray(object === null || object === void 0 ? void 0 : object.attributes) ? object.attributes.map(function (e) {
        return _attribute.Attribute.fromJSON(e);
      }) : []
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.owner !== undefined && (obj.owner = message.owner);
    message.auditor !== undefined && (obj.auditor = message.auditor);

    if (message.attributes) {
      obj.attributes = message.attributes.map(function (e) {
        return e ? _attribute.Attribute.toJSON(e) : undefined;
      });
    } else {
      obj.attributes = [];
    }

    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$owner, _object$auditor, _object$attributes;

    var message = createBaseProvider();
    message.owner = (_object$owner = object.owner) !== null && _object$owner !== void 0 ? _object$owner : "";
    message.auditor = (_object$auditor = object.auditor) !== null && _object$auditor !== void 0 ? _object$auditor : "";
    message.attributes = ((_object$attributes = object.attributes) === null || _object$attributes === void 0 ? void 0 : _object$attributes.map(function (e) {
      return _attribute.Attribute.fromPartial(e);
    })) || [];
    return message;
  }
};
exports.Provider = Provider;

function createBaseAuditedAttributes() {
  return {
    owner: "",
    auditor: "",
    attributes: []
  };
}

var AuditedAttributes = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.owner !== "") {
      writer.uint32(10).string(message.owner);
    }

    if (message.auditor !== "") {
      writer.uint32(18).string(message.auditor);
    }

    var _iterator2 = _createForOfIteratorHelper(message.attributes),
        _step2;

    try {
      for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
        var v = _step2.value;

        _attribute.Attribute.encode(v, writer.uint32(26).fork()).ldelim();
      }
    } catch (err) {
      _iterator2.e(err);
    } finally {
      _iterator2.f();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseAuditedAttributes();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.owner = reader.string();
          break;

        case 2:
          message.auditor = reader.string();
          break;

        case 3:
          message.attributes.push(_attribute.Attribute.decode(reader, reader.uint32()));
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      owner: (0, _helpers.isSet)(object.owner) ? String(object.owner) : "",
      auditor: (0, _helpers.isSet)(object.auditor) ? String(object.auditor) : "",
      attributes: Array.isArray(object === null || object === void 0 ? void 0 : object.attributes) ? object.attributes.map(function (e) {
        return _attribute.Attribute.fromJSON(e);
      }) : []
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.owner !== undefined && (obj.owner = message.owner);
    message.auditor !== undefined && (obj.auditor = message.auditor);

    if (message.attributes) {
      obj.attributes = message.attributes.map(function (e) {
        return e ? _attribute.Attribute.toJSON(e) : undefined;
      });
    } else {
      obj.attributes = [];
    }

    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$owner2, _object$auditor2, _object$attributes2;

    var message = createBaseAuditedAttributes();
    message.owner = (_object$owner2 = object.owner) !== null && _object$owner2 !== void 0 ? _object$owner2 : "";
    message.auditor = (_object$auditor2 = object.auditor) !== null && _object$auditor2 !== void 0 ? _object$auditor2 : "";
    message.attributes = ((_object$attributes2 = object.attributes) === null || _object$attributes2 === void 0 ? void 0 : _object$attributes2.map(function (e) {
      return _attribute.Attribute.fromPartial(e);
    })) || [];
    return message;
  }
};
exports.AuditedAttributes = AuditedAttributes;

function createBaseAttributesResponse() {
  return {
    attributes: []
  };
}

var AttributesResponse = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    var _iterator3 = _createForOfIteratorHelper(message.attributes),
        _step3;

    try {
      for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
        var v = _step3.value;
        AuditedAttributes.encode(v, writer.uint32(10).fork()).ldelim();
      }
    } catch (err) {
      _iterator3.e(err);
    } finally {
      _iterator3.f();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseAttributesResponse();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.attributes.push(AuditedAttributes.decode(reader, reader.uint32()));
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      attributes: Array.isArray(object === null || object === void 0 ? void 0 : object.attributes) ? object.attributes.map(function (e) {
        return AuditedAttributes.fromJSON(e);
      }) : []
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};

    if (message.attributes) {
      obj.attributes = message.attributes.map(function (e) {
        return e ? AuditedAttributes.toJSON(e) : undefined;
      });
    } else {
      obj.attributes = [];
    }

    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$attributes3;

    var message = createBaseAttributesResponse();
    message.attributes = ((_object$attributes3 = object.attributes) === null || _object$attributes3 === void 0 ? void 0 : _object$attributes3.map(function (e) {
      return AuditedAttributes.fromPartial(e);
    })) || [];
    return message;
  }
};
exports.AttributesResponse = AttributesResponse;

function createBaseAttributesFilters() {
  return {
    auditors: [],
    owners: []
  };
}

var AttributesFilters = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    var _iterator4 = _createForOfIteratorHelper(message.auditors),
        _step4;

    try {
      for (_iterator4.s(); !(_step4 = _iterator4.n()).done;) {
        var v = _step4.value;
        writer.uint32(10).string(v);
      }
    } catch (err) {
      _iterator4.e(err);
    } finally {
      _iterator4.f();
    }

    var _iterator5 = _createForOfIteratorHelper(message.owners),
        _step5;

    try {
      for (_iterator5.s(); !(_step5 = _iterator5.n()).done;) {
        var _v = _step5.value;
        writer.uint32(18).string(_v);
      }
    } catch (err) {
      _iterator5.e(err);
    } finally {
      _iterator5.f();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseAttributesFilters();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.auditors.push(reader.string());
          break;

        case 2:
          message.owners.push(reader.string());
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      auditors: Array.isArray(object === null || object === void 0 ? void 0 : object.auditors) ? object.auditors.map(function (e) {
        return String(e);
      }) : [],
      owners: Array.isArray(object === null || object === void 0 ? void 0 : object.owners) ? object.owners.map(function (e) {
        return String(e);
      }) : []
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};

    if (message.auditors) {
      obj.auditors = message.auditors.map(function (e) {
        return e;
      });
    } else {
      obj.auditors = [];
    }

    if (message.owners) {
      obj.owners = message.owners.map(function (e) {
        return e;
      });
    } else {
      obj.owners = [];
    }

    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$auditors, _object$owners;

    var message = createBaseAttributesFilters();
    message.auditors = ((_object$auditors = object.auditors) === null || _object$auditors === void 0 ? void 0 : _object$auditors.map(function (e) {
      return e;
    })) || [];
    message.owners = ((_object$owners = object.owners) === null || _object$owners === void 0 ? void 0 : _object$owners.map(function (e) {
      return e;
    })) || [];
    return message;
  }
};
exports.AttributesFilters = AttributesFilters;

function createBaseMsgSignProviderAttributes() {
  return {
    owner: "",
    auditor: "",
    attributes: []
  };
}

var MsgSignProviderAttributes = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.owner !== "") {
      writer.uint32(10).string(message.owner);
    }

    if (message.auditor !== "") {
      writer.uint32(18).string(message.auditor);
    }

    var _iterator6 = _createForOfIteratorHelper(message.attributes),
        _step6;

    try {
      for (_iterator6.s(); !(_step6 = _iterator6.n()).done;) {
        var v = _step6.value;

        _attribute.Attribute.encode(v, writer.uint32(26).fork()).ldelim();
      }
    } catch (err) {
      _iterator6.e(err);
    } finally {
      _iterator6.f();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseMsgSignProviderAttributes();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.owner = reader.string();
          break;

        case 2:
          message.auditor = reader.string();
          break;

        case 3:
          message.attributes.push(_attribute.Attribute.decode(reader, reader.uint32()));
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      owner: (0, _helpers.isSet)(object.owner) ? String(object.owner) : "",
      auditor: (0, _helpers.isSet)(object.auditor) ? String(object.auditor) : "",
      attributes: Array.isArray(object === null || object === void 0 ? void 0 : object.attributes) ? object.attributes.map(function (e) {
        return _attribute.Attribute.fromJSON(e);
      }) : []
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.owner !== undefined && (obj.owner = message.owner);
    message.auditor !== undefined && (obj.auditor = message.auditor);

    if (message.attributes) {
      obj.attributes = message.attributes.map(function (e) {
        return e ? _attribute.Attribute.toJSON(e) : undefined;
      });
    } else {
      obj.attributes = [];
    }

    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$owner3, _object$auditor3, _object$attributes4;

    var message = createBaseMsgSignProviderAttributes();
    message.owner = (_object$owner3 = object.owner) !== null && _object$owner3 !== void 0 ? _object$owner3 : "";
    message.auditor = (_object$auditor3 = object.auditor) !== null && _object$auditor3 !== void 0 ? _object$auditor3 : "";
    message.attributes = ((_object$attributes4 = object.attributes) === null || _object$attributes4 === void 0 ? void 0 : _object$attributes4.map(function (e) {
      return _attribute.Attribute.fromPartial(e);
    })) || [];
    return message;
  }
};
exports.MsgSignProviderAttributes = MsgSignProviderAttributes;

function createBaseMsgSignProviderAttributesResponse() {
  return {};
}

var MsgSignProviderAttributesResponse = {
  encode: function encode(_) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();
    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseMsgSignProviderAttributesResponse();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(_) {
    return {};
  },
  toJSON: function toJSON(_) {
    var obj = {};
    return obj;
  },
  fromPartial: function fromPartial(_) {
    var message = createBaseMsgSignProviderAttributesResponse();
    return message;
  }
};
exports.MsgSignProviderAttributesResponse = MsgSignProviderAttributesResponse;

function createBaseMsgDeleteProviderAttributes() {
  return {
    owner: "",
    auditor: "",
    keys: []
  };
}

var MsgDeleteProviderAttributes = {
  encode: function encode(message) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();

    if (message.owner !== "") {
      writer.uint32(10).string(message.owner);
    }

    if (message.auditor !== "") {
      writer.uint32(18).string(message.auditor);
    }

    var _iterator7 = _createForOfIteratorHelper(message.keys),
        _step7;

    try {
      for (_iterator7.s(); !(_step7 = _iterator7.n()).done;) {
        var v = _step7.value;
        writer.uint32(26).string(v);
      }
    } catch (err) {
      _iterator7.e(err);
    } finally {
      _iterator7.f();
    }

    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseMsgDeleteProviderAttributes();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.owner = reader.string();
          break;

        case 2:
          message.auditor = reader.string();
          break;

        case 3:
          message.keys.push(reader.string());
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(object) {
    return {
      owner: (0, _helpers.isSet)(object.owner) ? String(object.owner) : "",
      auditor: (0, _helpers.isSet)(object.auditor) ? String(object.auditor) : "",
      keys: Array.isArray(object === null || object === void 0 ? void 0 : object.keys) ? object.keys.map(function (e) {
        return String(e);
      }) : []
    };
  },
  toJSON: function toJSON(message) {
    var obj = {};
    message.owner !== undefined && (obj.owner = message.owner);
    message.auditor !== undefined && (obj.auditor = message.auditor);

    if (message.keys) {
      obj.keys = message.keys.map(function (e) {
        return e;
      });
    } else {
      obj.keys = [];
    }

    return obj;
  },
  fromPartial: function fromPartial(object) {
    var _object$owner4, _object$auditor4, _object$keys;

    var message = createBaseMsgDeleteProviderAttributes();
    message.owner = (_object$owner4 = object.owner) !== null && _object$owner4 !== void 0 ? _object$owner4 : "";
    message.auditor = (_object$auditor4 = object.auditor) !== null && _object$auditor4 !== void 0 ? _object$auditor4 : "";
    message.keys = ((_object$keys = object.keys) === null || _object$keys === void 0 ? void 0 : _object$keys.map(function (e) {
      return e;
    })) || [];
    return message;
  }
};
exports.MsgDeleteProviderAttributes = MsgDeleteProviderAttributes;

function createBaseMsgDeleteProviderAttributesResponse() {
  return {};
}

var MsgDeleteProviderAttributesResponse = {
  encode: function encode(_) {
    var writer = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _m0.Writer.create();
    return writer;
  },
  decode: function decode(input, length) {
    var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    var end = length === undefined ? reader.len : reader.pos + length;
    var message = createBaseMsgDeleteProviderAttributesResponse();

    while (reader.pos < end) {
      var tag = reader.uint32();

      switch (tag >>> 3) {
        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },
  fromJSON: function fromJSON(_) {
    return {};
  },
  toJSON: function toJSON(_) {
    var obj = {};
    return obj;
  },
  fromPartial: function fromPartial(_) {
    var message = createBaseMsgDeleteProviderAttributesResponse();
    return message;
  }
};
exports.MsgDeleteProviderAttributesResponse = MsgDeleteProviderAttributesResponse;
