// akashjs/main/proto/akash/audit/v1beta1/audit.registry.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.registry = exports.load = exports.MessageComposer = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _audit = require("./audit");

var registry = [["/akash.audit.v1beta1.MsgSignProviderAttributes", _audit.MsgSignProviderAttributes], ["/akash.audit.v1beta1.MsgDeleteProviderAttributes", _audit.MsgDeleteProviderAttributes]];
exports.registry = registry;

var load = function load(protoRegistry) {
  registry.forEach(function (_ref) {
    var _ref2 = (0, _slicedToArray2["default"])(_ref, 2),
        typeUrl = _ref2[0],
        mod = _ref2[1];

    protoRegistry.register(typeUrl, mod);
  });
};

exports.load = load;
var MessageComposer = {
  encoded: {
    signProviderAttributes: function signProviderAttributes(value) {
      return {
        type_url: "/akash.audit.v1beta1.MsgSignProviderAttributes",
        value: _audit.MsgSignProviderAttributes.encode(value).finish()
      };
    },
    deleteProviderAttributes: function deleteProviderAttributes(value) {
      return {
        type_url: "/akash.audit.v1beta1.MsgDeleteProviderAttributes",
        value: _audit.MsgDeleteProviderAttributes.encode(value).finish()
      };
    }
  },
  withTypeUrl: {
    signProviderAttributes: function signProviderAttributes(value) {
      return {
        typeUrl: "/akash.audit.v1beta1.MsgSignProviderAttributes",
        value: value
      };
    },
    deleteProviderAttributes: function deleteProviderAttributes(value) {
      return {
        typeUrl: "/akash.audit.v1beta1.MsgDeleteProviderAttributes",
        value: value
      };
    }
  },
  toJSON: {
    signProviderAttributes: function signProviderAttributes(value) {
      return {
        typeUrl: "/akash.audit.v1beta1.MsgSignProviderAttributes",
        value: _audit.MsgSignProviderAttributes.toJSON(value)
      };
    },
    deleteProviderAttributes: function deleteProviderAttributes(value) {
      return {
        typeUrl: "/akash.audit.v1beta1.MsgDeleteProviderAttributes",
        value: _audit.MsgDeleteProviderAttributes.toJSON(value)
      };
    }
  },
  fromJSON: {
    signProviderAttributes: function signProviderAttributes(value) {
      return {
        typeUrl: "/akash.audit.v1beta1.MsgSignProviderAttributes",
        value: _audit.MsgSignProviderAttributes.fromJSON(value)
      };
    },
    deleteProviderAttributes: function deleteProviderAttributes(value) {
      return {
        typeUrl: "/akash.audit.v1beta1.MsgDeleteProviderAttributes",
        value: _audit.MsgDeleteProviderAttributes.fromJSON(value)
      };
    }
  },
  fromPartial: {
    signProviderAttributes: function signProviderAttributes(value) {
      return {
        typeUrl: "/akash.audit.v1beta1.MsgSignProviderAttributes",
        value: _audit.MsgSignProviderAttributes.fromPartial(value)
      };
    },
    deleteProviderAttributes: function deleteProviderAttributes(value) {
      return {
        typeUrl: "/akash.audit.v1beta1.MsgDeleteProviderAttributes",
        value: _audit.MsgDeleteProviderAttributes.fromPartial(value)
      };
    }
  }
};
exports.MessageComposer = MessageComposer;
