// akashjs/main/proto/akash/audit/v1beta2/audit.amino.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AminoConverter = void 0;
var AminoConverter = {
  "/akash.audit.v1beta2.MsgSignProviderAttributes": {
    aminoType: "/akash.audit.v1beta2.MsgSignProviderAttributes",
    toAmino: function toAmino(_ref) {
      var owner = _ref.owner,
          auditor = _ref.auditor,
          attributes = _ref.attributes;
      return {
        owner: owner,
        auditor: auditor,
        attributes: attributes.map(function (el0) {
          return {
            key: el0.key,
            value: el0.value
          };
        })
      };
    },
    fromAmino: function fromAmino(_ref2) {
      var owner = _ref2.owner,
          auditor = _ref2.auditor,
          attributes = _ref2.attributes;
      return {
        owner: owner,
        auditor: auditor,
        attributes: attributes.map(function (el0) {
          return {
            key: el0.key,
            value: el0.value
          };
        })
      };
    }
  },
  "/akash.audit.v1beta2.MsgDeleteProviderAttributes": {
    aminoType: "/akash.audit.v1beta2.MsgDeleteProviderAttributes",
    toAmino: function toAmino(_ref3) {
      var owner = _ref3.owner,
          auditor = _ref3.auditor,
          keys = _ref3.keys;
      return {
        owner: owner,
        auditor: auditor,
        keys: keys
      };
    },
    fromAmino: function fromAmino(_ref4) {
      var owner = _ref4.owner,
          auditor = _ref4.auditor,
          keys = _ref4.keys;
      return {
        owner: owner,
        auditor: auditor,
        keys: keys
      };
    }
  }
};
exports.AminoConverter = AminoConverter;
