// akashjs/main/proto/akash/provider/v1beta2/provider.amino.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AminoConverter = void 0;
var AminoConverter = {
  "/akash.provider.v1beta2.MsgCreateProvider": {
    aminoType: "/akash.provider.v1beta2.MsgCreateProvider",
    toAmino: function toAmino(_ref) {
      var owner = _ref.owner,
          hostUri = _ref.hostUri,
          attributes = _ref.attributes,
          info = _ref.info;
      return {
        owner: owner,
        host_uri: hostUri,
        attributes: attributes.map(function (el0) {
          return {
            key: el0.key,
            value: el0.value
          };
        }),
        info: {
          email: info.email,
          website: info.website
        }
      };
    },
    fromAmino: function fromAmino(_ref2) {
      var owner = _ref2.owner,
          host_uri = _ref2.host_uri,
          attributes = _ref2.attributes,
          info = _ref2.info;
      return {
        owner: owner,
        hostUri: host_uri,
        attributes: attributes.map(function (el0) {
          return {
            key: el0.key,
            value: el0.value
          };
        }),
        info: {
          email: info.email,
          website: info.website
        }
      };
    }
  },
  "/akash.provider.v1beta2.MsgUpdateProvider": {
    aminoType: "/akash.provider.v1beta2.MsgUpdateProvider",
    toAmino: function toAmino(_ref3) {
      var owner = _ref3.owner,
          hostUri = _ref3.hostUri,
          attributes = _ref3.attributes,
          info = _ref3.info;
      return {
        owner: owner,
        host_uri: hostUri,
        attributes: attributes.map(function (el0) {
          return {
            key: el0.key,
            value: el0.value
          };
        }),
        info: {
          email: info.email,
          website: info.website
        }
      };
    },
    fromAmino: function fromAmino(_ref4) {
      var owner = _ref4.owner,
          host_uri = _ref4.host_uri,
          attributes = _ref4.attributes,
          info = _ref4.info;
      return {
        owner: owner,
        hostUri: host_uri,
        attributes: attributes.map(function (el0) {
          return {
            key: el0.key,
            value: el0.value
          };
        }),
        info: {
          email: info.email,
          website: info.website
        }
      };
    }
  },
  "/akash.provider.v1beta2.MsgDeleteProvider": {
    aminoType: "/akash.provider.v1beta2.MsgDeleteProvider",
    toAmino: function toAmino(_ref5) {
      var owner = _ref5.owner;
      return {
        owner: owner
      };
    },
    fromAmino: function fromAmino(_ref6) {
      var owner = _ref6.owner;
      return {
        owner: owner
      };
    }
  }
};
exports.AminoConverter = AminoConverter;
